### PR TITLE
Add Build Address gRPC Query

### LINKED ISSUE #1705: Add gRPC querier for Build Address Query
Right now, the Build Address query used for instantiate2 is exposed as a CLI query only: https://github.com/CosmWasm/wasmd/blob/62d91d972a913888eb1c6ef692e372b8ceb1f560/x/wasm/client/cli/query.go#L73

For my current use case, this requires all deployers of our contracts to have daemons for multiple networks locally installed to then generate the necessary instantiate2 addresses in advance.

I would love for the query to be implemented in the gRPC query service, allowing for easier/automatic deployment processes that need to pre-gen the addresses just using public endpoints.

If the team is in favor of this addition, I can write the PR.

### PR DESCRIPTION
**Pull request by NotJeremyLiu**
Closes #1705

### CODE REVIEW

**Comment by NotJeremyLiu** [comment]
Not exaclty suure why test-system CI is failing, running make test-system locally passes

**Comment by NotJeremyLiu** [comment]
@alpe this is ready for review whenever you get the chance

File: x/wasm/keeper/querier_test.go [hunk 1]
```diff
@@ -1047,3 +1047,60 @@ func TestEnsurePaginationParams(t *testing.T) {
 		})
 	}
 }
+
+func TestQueryBuildAddress(t *testing.T) {
+	specs := map[string]struct {
+		srcQuery *types.QueryBuildAddressRequest
```
Comment (pinosu) [inline x/wasm/keeper/querier_test.go:1053]:
```suggestion
		src *types.QueryBuildAddressRequest
```
personal preference, just to keep consistent with all the others

File: x/wasm/keeper/querier_test.go [hunk 2]
```diff
@@ -1047,3 +1047,60 @@ func TestEnsurePaginationParams(t *testing.T) {
...
+	ctx, keepers := CreateTestInput(t, false, AvailableCapabilities)
+	keeper := keepers.WasmKeeper
+
+	q := Querier(keeper)
+	for msg, spec := range specs {
+		t.Run(msg, func(t *testing.T) {
+			got, gotErr := q.BuildAddress(ctx, spec.srcQuery)
+			if spec.expErr != nil {
+				fmt.Println("here")
+				fmt.Println("err: ", gotErr)
```
Comment (pinosu) [inline x/wasm/keeper/querier_test.go:1096]:
```suggestion

```
please remove these

**Review by pinosu — CHANGES_REQUESTED** [review]
Thanks for taking this task! 💐 
There are some changes needed to improve code quality and also it would be great if you could add more test cases such as wrong args, missing args etc..
Overall really nice work! 💯

File: x/wasm/keeper/querier_test.go (hunk 2 above)
Comment (NotJeremyLiu) [reply x/wasm/keeper/querier_test.go]: whoops, apologies

**Comment by NotJeremyLiu** [comment]
> Thanks for taking this task! 💐 There are some changes needed to improve code quality and also it would be great if you could add more test cases such as wrong args, missing args etc.. Overall really nice work! 💯

Thank you for reviewing, will address feedback later tonight!

**Comment by pinosu** [comment]
@NotJeremyLiu thanks for making the changes. Now code looks good. 💯 
Could you modify the `GetCmdBuildAddress()` function to use the querier instead of calling directly the keeper?

**Comment by NotJeremyLiu** [comment]
> @NotJeremyLiu thanks for making the changes. Now code looks good. 💯 Could you modify the `GetCmdBuildAddress()` function to use the querier instead of calling directly the keeper?

@pinosu Yes I can do that, sorry have been busy with work, will ping you directly once done and ready for re-review!

**Comment by NotJeremyLiu** [comment]
> @NotJeremyLiu thanks for making the changes. Now code looks good. 💯 Could you modify the `GetCmdBuildAddress()` function to use the querier instead of calling directly the keeper?

Done here: https://github.com/CosmWasm/wasmd/pull/1753/commits/b17c9fbb8fbd7a617879f530f511604b77cedf21

File: x/wasm/keeper/querier_test.go (hunk 1 above)
Comment (NotJeremyLiu) [reply x/wasm/keeper/querier_test.go]: done here: https://github.com/CosmWasm/wasmd/pull/1753/commits/e405904b253c16e3cfd8191dfda5b63591ba565d

**Comment by NotJeremyLiu** [comment]
all feedback addressed and more unit tests added, ready for re-review

**Review by pinosu — APPROVED** [review]
Nice work! 💯 LGTM 👍

**Pull request merged**